log: report hostname when COCKROACH_REPORT_SENSITIVE_DETAILS=true

tbg opened the pull request:
This environment variable will be set on our internal clusters and nightly test clusters
to allow easier identification.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19928)
<!-- Reviewable:end -->

mberhault commented on pkg/util/log/crash_reporting.go:
```diff
@@ -66,6 +66,12 @@ var (
+	//
+	// This should not be used by anyone unwilling to share the whole cluster
+	// data with Cockroach Labs and various cloud services.
+	ReportUnanonymized = envutil.EnvOrDefaultBool("COCKROACH_REPORT_UNANONYMIZED", false)
```
unanymized.. nonymized... anon.. ugh! this is a mouthful. How about "uncensored", "unredacted", "..._SENSITIVE_DETAILS`, etc...

tschottdorf replied:
The first isn't correct, the second one isn't clear enough, I could do the third.

mberhault replied:
I think the third is clear and describes the danger pretty well. Bonus points for being easier to say.

tschottdorf replied:
Yup, done.

dt commented:
random idea: as long as we're adding a env var, we could make it a string instead with just a default value `<redacted>`, and then we can explicitly set it to just what we want without needing to worry about the hostname?

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/util/log/crash_reporting.go, line 74 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19928#-KySmB-wCytN8l_4-NOn-r1-74:-KyW7kSE84ItboipTIse:brfqkpc) ([raw file](https://github.com/cockroachdb/cockroach/blob/343a3b4adad0c1889c457726a7156213347b3b23/pkg/util/log/crash_reporting.go#L74)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

I think the third is clear and describes the danger pretty well. Bonus points for being easier to say.
</blockquote></details>

even more bike shed: I think the wording is too broad -- the diagnostic reporting is certainly still anonymized of scheme strings and such. maybe `COCKROACH_CRASH_REPORTING_NAME` or something?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19928#-:-KyW71BSV2kV1uxqmkP7:b-khr5lj)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
I considered that, but then it needs to be set to something useful and it's more likely than not that during copy-pasting we'll just have conflicting values. Hostname is usually clear and it can't be messed up during setup.

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/util/log/crash_reporting.go, line 74 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19928#-KySmB-wCytN8l_4-NOn-r1-74:-KyW8Ro11QVyvP701tQu:bhwodii) ([raw file](https://github.com/cockroachdb/cockroach/blob/343a3b4adad0c1889c457726a7156213347b3b23/pkg/util/log/crash_reporting.go#L74)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

even more bike shed: I think the wording is too broad -- the diagnostic reporting is certainly still anonymized of scheme strings and such. maybe `COCKROACH_CRASH_REPORTING_NAME` or something?
</blockquote></details>

My thinking here was that we may want to deanonymize more in the future, and that under this env var we would be continually free to do so. For example, if we want to report unredacted errors, we don't want to go through all of our infra and rename all vars. This is the "don't expect any privacy" flag. I think that's most useful given the intended usage.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19928#-:-KyW8eXPUYVdgyqjBv6g:b-yp4hzh)*
<!-- Sent from Reviewable.io -->

dt commented:
:thumbsup: 

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/util/log/crash_reporting.go, line 74 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/19928#-KySmB-wCytN8l_4-NOn-r1-74:-KyW9-hbecFY0PhwXJJJ:b4c9lyr) ([raw file](https://github.com/cockroachdb/cockroach/blob/343a3b4adad0c1889c457726a7156213347b3b23/pkg/util/log/crash_reporting.go#L74)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

My thinking here was that we may want to deanonymize more in the future, and that under this env var we would be continually free to do so. For example, if we want to report unredacted errors, we don't want to go through all of our infra and rename all vars. This is the "don't expect any privacy" flag. I think that's most useful given the intended usage.
</blockquote></details>

:thumbsup: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19928#-:-KyW91KOIGAjglFUlKqC:b4c9lyr)*
<!-- Sent from Reviewable.io -->

tbg committed:
log: report hostname when COCKROACH_REPORT_SENSITIVE_DETAILS=true

This environment variable will be set on our internal clusters and nightly test
clusters to allow easier identification.

tschottdorf commented:
TFTRs!

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19928#-:-KyWAY1SHYlWRT9WJhXN:bz5mair)*
<!-- Sent from Reviewable.io -->
